internal/util/resourceutil/id.go

```diff
@@ -27,9 +27,9 @@ func ParseID(value types.String) (int, diag.Diagnostics) {
 			"ID is larger than your system supports.",
 			"The current version of the provider has problems with IDs > 32bit on 32 bit systems. If possible, switch to a 64 bit system for now. See https://github.com/hetznercloud/hcloud-go/issues/263")
 	}
-	return int(id), diagnostics
+	return id, diagnostics
```

Comment:
The check above this line (`id > math.MaxInt32 && strconv.IntSize == 32`) can be removed, it was only relevant for the `int(id)` cast.